modules/exploits/multi/postgres/postgres_cmd_execution_nine_three.rb

```diff
@@ -0,0 +1,250 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/exploit/postgres'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Postgres
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'PostgreSQL >9.3 Command Execution',
+      'Description' => %q(
+        Installations running Postgres 9.3 and above have functionality which allows for the superuser 
+        and users with 'pg_read_server_files' to pipe to and from an external program using COPY. 
+        This allows arbitary command execution as though you have console access.
+
+        This module attempts to create a new table, then execute system commands in the context of 
+        copying the command output into the table.
+
+        This module should work on all Postgres systems running version 9.3 and above. 
+
+        For Linux & OSX systems, cmd stagers are used such as: cmd/unix/reverse_perl
+
+        For Windows Systems command is placed in the COMMAND parameter, such as a powershell 
+        download cradle for meterpreter
+      ),
+      'Author' => [
+        'Jacob Wilkin', # author of this module
+        'Micheal Cottingham', # the postgres_createlang module that this is based on
+      ],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['URL', '<Blogpost on the subject>'],
+        ['URL', 'https://www.postgresql.org/docs/9.3/release-9-3.html'] #Patch notes adding the function, see 'E.26.3.3. Queries - Add support for piping COPY and psql \copy data to/from an external program (Etsuro Fujita)'
+      ],
+        'PayloadType' => %w(cmd)
+      'Platform' => %w(linux unix win osx),
+      'Payload' => {
+      },
+      'Arch' => [ARCH_CMD],
+      'Targets' => [
+        ['Automatic', {}]
+      ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Feb 24 2019'
```

Comment:
This is usually the date the technique was published, or the date the vulnerable functionality was introduced.

Reply:
Fixed